File: 2019s/HellsImmigration/HellsImmigration.java
import java.io.*;
import java.util.*;

public class HellsImmigration {
    // do not change anything in main method
    public static void main(String[] args) {
		try {
			BufferedReader br = new BufferedReader(new FileReader("HellsImmigrationIN.txt"));
			int testCases = Integer.parseInt(br.readLine());
			
			for (int i = 0; i < testCases; i++) {
                String[] line = br.readLine().split(" ");
                int N = Integer.parseInt(line[0]);
                int M = Integer.parseInt(line[1]);

                List<Shift> shift = new ArrayList<> ();
				for (int j = 0; j < N; j++) {
                    line = br.readLine().split(" ");
                    shift.add(new Shift(Integer.parseInt(line[0]), Integer.parseInt(line[1])));
                }
                List<Edge> edges = new ArrayList<> ();
                for (int j = 0; j < M; j++) {
                    line = br.readLine().split(" ");
                    edges.add(new Edge(Integer.parseInt(line[0]), Integer.parseInt(line[1]), Integer.parseInt(line[2])));
                }
                line = br.readLine().split(" ");
                int s = Integer.parseInt(line[0]);
                int d = Integer.parseInt(line[1]);
				System.out.println(helpRordon(N, M, shift, edges, s, d));
            }
            br.close();
		} catch (Exception e) {
			e.printStackTrace();
		}
    }

    public static int helpRordon(int n, int m, List<Shift> shift, List<Edge> edges, int s, int d) {
       return -1;
    }

    static class Shift {
        public int first;
        public int second;

        public Shift(int first, int second) {
            this.first = first;
            this.second = second;
        }
    }

    static class Edge {
        public int u;
        public int v;
        public int weight;

        public Edge(int u, int v, int weight) {
            this.u = u;
            this.v = v;
            this.weight = weight;
        }
    }
}

File: 2019s/HellsImmigration/HellsImmigration.py
def helpRordan(n, m, shift, edges, s, d):
  pass

if __name__ == '__main__':
  with open('HellsImmigrationIN.txt', 'r') as f:
    num_test_cases = int(f.readline().strip())
    for i in range(num_test_cases):
      n, m = f.readline().strip().split()
      n, m = int(n), int(m)
      shift = []
      edges = []
      for i in range(0, n):
        first, second = f.readline().strip().split()
        first, second = int(first), int(second)
        shift.append((first, second))
        
      
      for i in range(0, m):
        b, e, w = f.readline().strip().split()
        b, e, w = int(b), int(e), int(w)
        edges.append((b,e,w))
        
      
      s, d = f.readline().strip().split()
      s, d = int(s), int(d)
      print(helpRordan(n, m, shift, edges, s, d))
      